schemas/com.snowplowanalytics.snowplow/client_session/jsonschema/1-0-3

```diff
@@ -0,0 +1,82 @@
+{
+	"$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+	"$supersedes": ["1-0-2"],
```

Comment:
First time we've done this in Iglu Central!  Which is exciting and terrifying.

Have you checked the dev registry created by this PR?

```
http://iglucentral-dev.com.s3-website-us-east-1.amazonaws.com/BRANCH_NAME
```

It would be great if you could confirm:

- If you fetch the 1-0-3 schema from the dev registry then it includes the `$supersedes` field
- If you fetch the 1-0-2 schema from the dev registry then it includes the `$supersededBy` field.

The dev registry is built in the same way as the main iglu-central registry.  If it works on the dev registry then it will also work in Iglu Central when this PR gets merged.

Reply:
Good point! Altough the 1-0-3 schema contains the `$supersedes` field, it seems that the `$supersededBy` field is not added to the 1-0-2 schema.

Here are links to the schemas:

* http://iglucentral-dev.com.s3-website-us-east-1.amazonaws.com/issue/1410-update_client_sesison/schemas/com.snowplowanalytics.snowplow/client_session/jsonschema/1-0-2
* http://iglucentral-dev.com.s3-website-us-east-1.amazonaws.com/issue/1410-update_client_sesison/schemas/com.snowplowanalytics.snowplow/client_session/jsonschema/1-0-3

I will check with Nick if he has any idea why.

Reply:
So as discussed with Nick on Slack, I have removed the `$supersedes` relation from the `1-0-3` schema and added a `$supersededBy` relation to the `1-0-2` schema. Have tested this by pointing a Micro to the dev Iglu Server and it has worked – validated using the new schema.

Reply:
Hi @matus-tomlein sorry this is not quite right still.  You need the `$supersedes` relation in `1-0-3` AND the `$supersededBy` relation in `1-0-2`.  Sorry to disagree with what you got told in Slack!!

The reason is... for Iglu Central we push schemas to two different types of registry.  Two of them are static registries in S3:

- `http://iglucentral.com`
- `http://mirror01.iglucentral.com`

And two of them are Iglu Servers:

- `https://com-iglucentral-eu1-prod.iglu.snplow.net`
- `https://com-iglucentral-eu2-prod.iglu.snplow.net`

For the Iglu Servers it is important that the newer schema contains the `$supersedes` field or else it drops the information.

I just tested these schemas on a local Iglu Server, and I'm fairly sure everything will be ok with pushing these schemas.  However, we should definitely flag to Josh that there is a small element of risk because it's the first time we've done this with the Iglu Central Iglu Servers.

Reply:
Thanks a lot for validating that, @istreeter! I have added the `$supersedes` relation to the `1-0-3` schema back in.

Reply:
Well done for battling through this :)  If this works out well, then it sets a nice pattern for future PRs.  Hopefully it won't be so difficult next time!